pkg/internalapis/featurestates/featurestates.go

```diff
@@ -0,0 +1,368 @@
+		log.Errorf("failed to get kubeconfig. Error: %v", err)
+		return err
+	}
+	// create controller runtime client
+	controllerRuntimeClient, err = k8s.NewClientForGroup(ctx, config, CRDGroupName)
+	if err != nil {
+		log.Errorf("failed to create controllerRuntimeClient. Err: %v", err)
+		return err
+	}
+
+	// create/update feature state CRs in all workload namespaces
+	err = updateFeatureStatesCRInAllWorkLoadNamespaces(ctx, []featurestatesv1alpha1.FeatureState{})
+	if err != nil {
+		log.Errorf("failed to update feature states CR in the all workload namespaces. err: %v", err)
+		return err
+	}
+
+	// Create k8s Informer and watch on configmaps and namespaces
+	informer := k8s.NewInformer(k8sClient)
+	informer.AddConfigMapListener(ctx, k8sClient, svFeatureStateConfigMapNamespace,
+		// Add
+		nil,
+		// Update
+		func(oldObj interface{}, newObj interface{}) {
+			configMapUpdated(oldObj, newObj)
+		},
+		// Delete
+		nil)
+	informer.AddNamespaceListener(
+		// Add
+		func(obj interface{}) {
+			namespaceAdded(obj)
+		},
+		// Update
+		nil,
+		// Delete
+		nil)
+	informer.Listen()
+	log.Infof("informer on config-map and namespaces started")
+
+	// Start Retry routine to process failed create or update CRs
+	go pendingCRUpdatesObj.startRetryAttempts()
+	log.Infof("started background routine to process failed CR updates at regulat interval")
+	log.Infof("SvFSSReplicationService is running")
+	var stopCh = make(chan bool)
+	<-stopCh
+	return nil
+}
+
+// startRetryAttempts helps process pending failed CR updates
+func (pendingCRUpdatesObj *pendingCRUpdates) startRetryAttempts() {
+	for {
+		func() {
+			pendingCRUpdatesObj.lock.Lock()
+			defer pendingCRUpdatesObj.lock.Unlock()
+			ctx, log := logger.GetNewContextWithLogger()
+			log.Infof("performing retry operations on failed feature states CRs")
+
+			for namespace, featurestatesSpec := range pendingCRUpdatesObj.pendingCRUpdate {
+				if featurestatesSpec != nil {
+					// Check if namespace for the CR is still present on the cluster
+					_, err := k8sClient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+					if err != nil {
+						if apierrors.IsNotFound(err) {
+							// if Namespace is not present, mark pending task as done
+							pendingCRUpdatesObj.pendingCRUpdate[namespace] = nil
+							continue
+						} else {
+							log.Errorf("failed to check if namespace: %q is present on the cluster", namespace)
+							continue
+						}
+					}
+					// check if CR is present on the namespace
+					featurestateCR := &featurestatesv1alpha1.CnsCsiSvFeatureStates{}
+					err = controllerRuntimeClient.Get(ctx, client.ObjectKey{Name: SVFeatureStateCRName, Namespace: namespace}, featurestateCR)
+					if err != nil {
+						if !apierrors.IsNotFound(err) {
+							// attempt to Create the CR
+							err = controllerRuntimeClient.Create(ctx, featurestatesSpec)
+						}
+					} else {
+						// Attempt to Update CR
+						err = controllerRuntimeClient.Update(ctx, featurestatesSpec)
```

Comment:
Can we remove the `else` and keep the update outside?

Reply:
we need to keep update inside, as, after create, we do not want to apply update.